src/cron/schedule-code-review.js

```
@@ -0,0 +1,124 @@
+    return null;
+  }
+  return statusText;
+};
+
+/**
+ * Checks threads for a specific status and sends reminders in all text channels.
+ *
+ * @param {Client} client - The Discord.js client instance.
+ * @param {string} statusText - The status to look for in thread titles.
+ */
+const checkThreadsForReview = async (client, statusText) => {
+  try {
+    const guild = await client.guilds.fetch(DISCORD_SERVER.discordGuildId);
+
+    if (!guild) {
+      console.error(
+        `Guild with ID ${DISCORD_SERVER.discordGuildId} not found.`
+      );
+      return;
+    }
+
+    const channels = await guild.channels.fetch();
+    const textChannels = channels.filter(
+      (channel) => channel.type === ChannelType.GuildText
+    );
+
+    if (textChannels.size === 0) {
+      console.error('No text channels found.');
+      return;
+    }
+
+    for (const channel of textChannels.values()) {
+      try {
+        const threads = await channel.threads.fetchActive();
+        const pendingReviews = threads.threads.filter((thread) =>
+          thread.name.includes(statusText)
+        );
+
+        if (pendingReviews.size > 0) {
+          let messageContent = `🔔 **Pending Threads in ${channel.name}**:\n\n`;
+          for (const thread of pendingReviews.values()) {
+            messageContent += `The thread [${thread.name}] has not been reviewed:\n${thread.url}\n`;
+          }
+
+          await channel.send({
+            content: messageContent,
+          });
+        } else {
+          console.log(
+            translateLanguage('checkReview.noPendingReviews').replace(
+              '{{channelName}}',
+              channel.name
+            )
+          );
+        }
+      } catch (channelError) {
+        console.error(
+          `Error processing channel ${channel.name}: ${channelError.message}`
+        );
+      }
+    }
+  } catch (error) {
+    console.error(
+      `Error fetching guild or processing threads: ${error.message}`
+    );
+  }
```

In this case the bot would fail silently?

---

yes

---

well as this is a cronjob and we have not channel for reporting errors it is ok.

@joset98 I remember you added a logic for writing errors in a log file, is right?, can we do something to write these error logs in that file?, or am I confused and that was never done?

---

You could check here
>https://github.com/TeamNovaSoft/discord-bot/blob/94c83687df6ffc8e08ef01f4dfd8909e9ae75cd5/src/index.js#L66-L68